core/src/main/java/io/substrait/expression/Expression.java

```diff
@@ -684,41 +717,6 @@ public <R, E extends Throwable> R accept(ExpressionVisitor<R, E> visitor) throws
     }
   }
 
-  @Value.Immutable
-  abstract static class Window implements Expression {
-    @Nullable
-    public abstract Aggregate.Measure aggregateFunction();
-
-    @Nullable
-    public abstract WindowFunction windowFunction();
-
-    public abstract List<Expression> partitionBy();
-
-    public abstract List<SortField> orderBy();
-
-    public abstract WindowBound lowerBound();
-
-    public abstract WindowBound upperBound();
-
-    public abstract boolean hasNormalAggregateFunction();
-
-    public static ImmutableExpression.Window.Builder builder() {
-      return ImmutableExpression.Window.builder();
-    }
-
-    public <R, E extends Throwable> R accept(ExpressionVisitor<R, E> visitor) throws E {
-      return visitor.visit(this);
-    }
-  }
-
-  @Value.Immutable
-  public abstract static class WindowFunction {
-    public abstract WindowFunctionInvocation getFunction();
-
-    public abstract Optional<Expression> getPreMeasureFilter();
```

Comment:
The `preMeasureFilter` field was not used anywhere, and the [WindowFunction](https://github.com/substrait-io/substrait/blob/51765cccd92fa0014e9cf077cc8cc952731f33d0/proto/substrait/algebra.proto#L847) message cannot actually encode a pre-measure filter (for now).

I opted to remove it entirely. Support can be added in once the spec supports it.